src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLanguageServerClient.cs

```diff
@@ -116,7 +116,7 @@ public AutoFlushingStream(Stream inner)
 
             public override long Position { get => _inner.Position; set => _inner.Position = value; }
 
-            public override void Flush() => _inner.Flush();
+            public override void Flush() => _inner.FlushAsync();
```

Comment:
This looks like a crime. @NTaylorMullen I see [you suggested](https://github.com/dotnet/aspnetcore-tooling/pull/1737/files?w=1#r401243125) we do this. But why? This looks like it defeats the purpose of an `AutoFlushingStream`. This could lead to multiple writes happening without any flushes in between. Why not try-catch around `Flush()` instead if we don't want to crash VS?

Reply:
We care that flush happens eventually for the stream. The problem the auto flusher was intending to fix was where O# wouldn't flush after it would write ever. We still do a `FlushAsync` after every write below as well. Maybe I didn't understand the question but I don't know how a write ever happens without a "soon after" flush taking place.

Reply:
Would it help if I included a comment like I did below?

Also, I'm just happy that people are using my terminology and calling things like this "crimes" now lol.

Reply:
lol one day we're gonna worry all the hire ups when they overhear us all talking about how many crimes we've committed lmao.

Ya a comment's a great idea!